core/src/test/java/test/failedreporter/issue2517/FailedDataProviderWithFactoryReporterSampleTest.java

```diff
@@ -0,0 +1,31 @@
+package test.failedreporter.issue2517;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+public class FailedDataProviderWithFactoryReporterSampleTest {
+    private Integer data;
+
+    public FailedDataProviderWithFactoryReporterSampleTest() {
```

Comment:
We can remove this default constructor and make the `@DataProvider` static (I think you added this constructor because TestNG couldnt reach your data provider)

Reply:
Thank you @krmahadevan, It works.